src/transformers/generation/utils.py

```diff
@@ -351,6 +359,43 @@ def prepare_inputs_for_generation(self, *args, **kwargs):
             "A model class needs to define a `prepare_inputs_for_generation` method in order to use `.generate()`."
         )
 
+    def switch_cache_implementation(self, cache_implementation: Union[CacheImplementation, str], **kwargs):
```

Comment:
In practice, we are obviously always switching from one cache implementation to another, but for the users it's likely more intuitive to simply `set` a new cache implementation, so `set_cache_implementation` might be a better method.

Reply:
I agree! thanks for ponting that out !